docs/android/codestyle.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "Android Code Style"
+sidebar_label: "Code Style"
+---
+
+## Why Enforcing Code Style
+
+We aim to maintain a **consistent and standard codebase**. By enforcing code style:
+
+- We reduce unnecessary comments on PRs, allowing reviewers to focus on the logic rather than formatting.
+- We ensure the code is easier to read and maintain.
+
+## Language Guidelines
+
+- All code must be written in **English**.
+- Avoid typos and grammatical mistakes. While mistakes are acceptable (as many contributors are non-native speakers), reviewers are encouraged to suggest corrections.
+- Use tools like LLMs or spell checkers to assist in correcting mistakes.
+
+## KTLint Formatter
+
+We use [KTLint](https://pinterest.github.io/ktlint) to enforce Kotlin code style. It is integrated into our Gradle modules and configured via the `.editorconfig` file.
+
+### Custom Rules
+
+- We override some KTLint rules when necessary. To enable or disable a rule:
```

Comment:
Some? Which ones? If you're going to introduce a new process, why not require linking the exceptions here?

Reply:
You're introducing a process here and require reasoning. Looking back through PRs to find out what changed can be time consuming and not everyone knows about git blame.

So suggestion to add to the process: you also need to link the override in the docs, so that they are in one place with a human-readable title?

Reply:
So you would like in this docs all the overrides that we uses ? Otherwise we can force ppl to link it to a Github issue?

Reply:
Maybe? It just raised questions for me, if you want a "proper explanation" it feels like it should be more than a PR so I can get a feeling for what is considered necessary. Could just be me.

Unrelated change for this line: there is a dash here but only one list item:
```suggestion
We override some KTLint rules when necessary. To enable or disable a rule:
```